optimizely/event/dispatcher.go

```diff
@@ -20,11 +23,26 @@ type Dispatcher interface {
 type HTTPEventDispatcher struct {
 }
 
-var dispatcherLogger = logging.GetLogger("EventDispatcher")
-
 // DispatchEvent dispatches event with callback
 func (*HTTPEventDispatcher) DispatchEvent(event LogEvent, callback func(success bool)) {
+	dispatchEvent(event, callback)
+}
+
+type QueueEventDispatcher struct {
+	q Queue
+	mux sync.Mutex
+}
+
+func (ed *QueueEventDispatcher) DispatchEvent(event LogEvent, callback func(success bool)) {
+
+	ed.q.Add(event)
+	callback(true)
+	go func() {
+		ed.flushEvents()
+	}()
+}
 
+func dispatchEvent(event LogEvent, callback func(success bool)) {
```

Comment:
Having this static method might make it hard to unit test. Maybe we can try having the QueueEventDispatcher wrap around the HttpEventDispatcher and call the HttpEventDispatcher when it flushes?